widget/embedded/src/containers/Wallets/Wallets.tsx

```diff
@@ -92,17 +94,17 @@ function Main(props: PropsWithChildren<PropTypes>) {
         }
       }
     }
-    if (event === Events.ACCOUNTS && state.connected) {
+    if (
+      (event === Events.ACCOUNTS && state.connected) ||
+      // Hub works differently, and this check should be enough.
+      (event === Events.ACCOUNTS && meta.isHub)
```

Comment:
Legacy is firing only one batched `Events.ACCOUNTS`, but hub works differently and firing an event per namespace. It's important to this condition block be run on each namespace since as you can see `widget/embedded/src/hooks/useBootstrap/useBootstrap.ts:62`, we are storing last connected wallets for `autoconnect`.

`isHub` is a flag that has been added in `wallets-react` to distinguish event sources.